### ISSUE TITLE
Is this Software still Working?

### ISSUE THREAD

**Issue by WVkirai** [chris124567/hulu#6]
Thank you for this excellent piece of software. If you don’t mind me asking, I have a few questions. 

1)	Does this program still work as of today? I know that Google recently revoked their old Windows keys.
2)	From my understanding, this program works by emulating the functionality of widevinecdm.so rather than hooking the functions of the binary. Is this correct?
3)	Is the 'Android SDK built for x86' to which you refer an AVD on a 64-bit machine?

Thanks.
-WV

**Comment by chris124567** [comment]
1. Yes.  The keys Google revoked were for old Chrome versions.  I'm using a key from an Android device.  It could be revoked eventually but there any plenty of other keys around to replace it.
2. Yes.
3. I think so but I don't know for sure because I just found this key by searching Github.

**Comment by WVkirai** [comment]
Thank you for your reply.

In that case, the RSA keys are tied to a Widevine CDM embedded in Android OS, rather than in the Chrome app. This makes sense as “Widevine Content Decryption Module” is not listed as a Chrome component on Android. However, I tried this key out on an Android 11.0 x86 Pixel AVD and it didn’t work.

Would you be able to share a Github link to the source of the RSA key? No matter what keys words I use related to pywidevine, I still can’t seem to find anything of use.

Thanks.
-WV

**Comment by chris124567** [comment]
What are you talking about? You can run the software on any operating system.  The keys do not have to line up with what OS you are using.  The key just signs the license request to the widevine server.  Link: https://github.com/Gryffindor8/Netflix-videos-downloader/tree/main/pywidevine/cdm/devices/android_general

**Comment by WVkirai** [comment]
The reason why I asked about the origin of the key is because I intended to put the RSA key inside of widevine-L3-decryptor. My ultimate goal was to decrypt Widevine encrypted videos from a different site. But after that proved fruitless, I instead looked at your code and saw how easily modifiable it was. The only real difficulty I had was on the implementation of the PSSH box. For whatever reason, the website I was downloading videos from embedded the PSSH box not in the MPD manifest, but in the mp4 files themselves. But once I manually extracted them the rest was trivial, and I got the videos I wanted.

Thanks for making the program, it certainly helped me out.

Widevine really is dead.
-WV
